lib/src/search/widget/search_settings_widget.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:flutter/material.dart';
+
+import '../settings_controller.dart';
+
+const _selectedColor = Colors.black;
+const _unselectedColor = Color.fromARGB(88, 0, 0, 0);
+const _hintText = 'Search...';
+const _iconSize = 24.0;
+
+class SearchSettingsWidget extends StatefulWidget {
+  final FocusNode focusNode;
+  final SearchSettingsController settingsController;
+
+  const SearchSettingsWidget({
+    super.key,
+    required this.focusNode,
+    required this.settingsController,
+  });
+
+  @override
+  State<SearchSettingsWidget> createState() => _SearchSettingsWidgetState();
+}
+
+class _SearchSettingsWidgetState extends State<SearchSettingsWidget> {
+  late final focusNode = widget.focusNode;
+  late final settingsController = widget.settingsController;
+  bool _isCaseSensitive = false;
+  bool _isRegex = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: Padding(
+            padding: const EdgeInsets.all(8),
+            child: TextField(
+              decoration: const InputDecoration(
+                hintText: _hintText,
+                isCollapsed: true,
+                border: InputBorder.none,
+              ),
+              focusNode: focusNode,
+              enabled: true,
+              controller: settingsController.patternController,
+            ),
+          ),
+        ),
+        InkWell(
+          hoverColor: Colors.transparent,
+          onTap: () {
+            setState(() {
+              _isCaseSensitive = !_isCaseSensitive;
+              settingsController.value = settingsController.value.copyWith(
+                isCaseSensitive: _isCaseSensitive,
+              );
+            });
+          },
```

Comment:
Use ToggleButtons, then:
```suggestion
          onTap: (index) {
            // TODO: Use keyed_collection_widgets when this lands: https://github.com/alexeyinkin/flutter-keyed-collection-widgets/issues/2
            switch (index) {
              case 0:
                settingsController.toggleCaseSensitivity();
                break;
              case 1:
                settingsController.toggleRegExp();
                break;
            }
          },
```